Scenarios/AKS-Secure-Baseline-PrivateCluster/Terraform/05-Network-LZ/app-gateway.tf

```diff
@@ -23,33 +60,45 @@ resource "azurerm_subnet_network_security_group_association" "appgwsubnet" {
   network_security_group_id = module.appgw_nsgs.appgw_nsg_id
 }
 
+# based on the structure of the appgws map are deployed multiple appplication gateway, usually this is used in the blue green scenario
 resource "azurerm_public_ip" "appgw" {
-  name                = "appgw-pip"
+  for_each = { for appgws in local.appgws : appgws.name_prefix => appgws if appgws.appgw_turn_on == true}
```

Comment:
Please add comment indicating that if Blue/Green scenario is not used then this code can be simplified by removing the for_each look and using a single app gateway.